src/main/java/onlinebookstore/dto/user/UserRegistrationRequestDto.java

```diff
@@ -40,6 +42,10 @@ public record UserRegistrationRequestDto(
         String lastName,
 
         @Size(max = 255, message = "Shipping address must not exceed 255 characters.")
-        String shippingAddress
+        String shippingAddress,
+
+        @NotEmpty(message = "Invalid roles. Roles shouldn't be empty or null.")
+        Set<String> roles
+
```

Comment:
remove this field. It’s  not secure that user can choose a  role during registration